tests/impl.cpp

```diff
@@ -9149,14 +9149,71 @@ result_t test_mm_cmpestrc(const SSE2NEONTestImpl &impl, uint32_t iter)
     return TEST_UNIMPL;
 }
 
+typedef struct {
+    int16_t a[8], b[8];
+    int la, lb;
+    const int imm8;
+    int expect;
+} test_mm_cmpestri_data_t;
+static test_mm_cmpestri_data_t test_mm_cmpestri_data = {
+    {13, 6, 5, 4, 3, 2, 1, 3},
+    {-7, 16, 5, 4, -1, 6, 1, 3},
+    10,
+    10,
+    _SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT,
+    7,
+};
+
 result_t test_mm_cmpestri(const SSE2NEONTestImpl &impl, uint32_t iter)
 {
-    return TEST_UNIMPL;
-}
+    int16_t *a = test_mm_cmpestri_data.a, *b = test_mm_cmpestri_data.b;
+    int la = test_mm_cmpestri_data.la, lb = test_mm_cmpestri_data.lb;
+    const int imm8 = test_mm_cmpestri_data.imm8;
+    int expect = test_mm_cmpestri_data.expect;
+    __m128i ma, mb;
+    memcpy(&ma, a, sizeof(ma));
+    memcpy(&mb, b, sizeof(mb));
+#if defined(__arm__) || defined(__aarch64__)
```

Comment:
Can you avoid the special cases (Arm/Aarch64) for invoking `_mm_cmpestri`?

Reply:
Conditional build with `imm` rather than intrinsic call. i.e., 
```c
#if defined(__arm__) || defined(__aarch64__)
#define IMM (something)
#else
#define IMM (_SIDD_SWORD_OPS | _SIDD_CMP_RANGES | _SIDD_MOST_SIGNIFICANT)
```